### PR TITLE
Implement cookiecutter-plone-starter templates in cookieplone-templates

### LINKED ISSUE #2: Create monorepo template (projects, addons)
# Goals

- [ ] Migrate [cookiecutter-plone-starter](https://github.com/collective/cookiecutter-plone-starter) to cookieplone-templates

# Blockers

- [ ] Usage of `pre-commit` / `husky` / `plone/meta` in a monorepo

### PR DESCRIPTION
**Pull request by ericof**
Fixes #2

### CODE REVIEW

**Comment by jensens** [comment]
cookieplone question: how can I point cookieplone to a branch of the cookieplone-templates like this one?

**Comment by sneridagh** [comment]
> cookieplone question: how can I point cookieplone to a branch of the cookieplone-templates like this one?

@jensens you can do it with a local checkout, using the env var of cookieplone as in:

`COOKIEPLONE_REPOSITORY=/Users/sneridagh/Development/plone/cookiecutter-plone pipx run cookieplone`

**Comment by jensens** [comment]
Findings:
- Plone Version wrong at least in `constraints.txt` and `MakeFile`
- We still have a `setup.py` it is not needed any more these days and looks ancient -> move all in pyproject.toml

**Comment by jensens** [comment]
`mxdev` can be run with `pipx` as well as probably `pre-commit` and `tox` can.

**Comment by jensens** [comment]
> The top-level is used to make tags for the project. @fredvd came up with this iirc.

`backend/version.txt` is what I meant. And it contains a Plone version number. I doubt we need it.
+1 for a way to define a top-level version for the whole construct of frontend/backend/devops. Not sure if the top-level `version.txt` is the way to go.

**Comment by sneridagh** [comment]
@jensens we need that for the Makefile to get the current Plone version (PLONE_VERSION) to pass it to the images.

ok, now I see that has not been transferred from plone-starter to here.

https://github.com/collective/cookiecutter-plone-starter/blob/main/%7B%7B%20cookiecutter.project_slug%20%7D%7D/backend/Makefile#L20

https://github.com/collective/cookiecutter-plone-starter/blob/main/%7B%7B%20cookiecutter.project_slug%20%7D%7D/backend/Dockerfile#L2

So, on upgrade, we only have to update it once in one place. With the exception of the constraints version, which is another story, though.

File: project/README.md [hunk 1]
```diff
@@ -45,13 +45,13 @@ Cookieplone will ask a lot of questions.
 You can use some of its options to avoid repeatedly entering the same values.
 
 
-#### `--no-input`
+#### `--no_input`
 
-You can use the [`--no-input`](https://cookiecutter.readthedocs.io/en/latest/cli_options.html#cmdoption-cookiecutter-no-input) option to make the cookiecutter not prompt for parameters and only use `cookiecutter.json` file content.
+You can use the [`--no_input`](https://cookiecutter.readthedocs.io/en/latest/cli_options.html#cmdoption-cookiecutter-no-input) option to make the cookieplone not prompt for parameters and only use `cookiecutter.json` file content.
```
Comment (stevepiercy) [inline project/README.md:50]:
```suggestion
You can use the [`--no_input`](https://cookiecutter.readthedocs.io/en/latest/cli_options.html#cmdoption-cookiecutter-no-input) option to make cookieplone not prompt for parameters and only use `cookiecutter.json` file content.
```

File: project/README.md [hunk 2]
```diff
@@ -45,13 +45,13 @@ Cookieplone will ask a lot of questions.
...
 
 
 #### `--replay` and `--replay-file`
-You can use the options [`--replay`](https://cookiecutter.readthedocs.io/en/latest/cli_options.html#cmdoption-cookiecutter-replay) and [`--replay-file`](https://cookiecutter.readthedocs.io/en/latest/cli_options.html#cmdoption-cookiecutter-replay-file) to not prompt for parameters and only use information entered previously or from a specified file.
+You can use the option [`--replay-file`](https://cookiecutter.readthedocs.io/en/latest/cli_options.html#cmdoption-cookiecutter-replay-file) to not prompt for parameters and only use information entered previously or from a specified file.
```
Comment (stevepiercy) [inline project/README.md:54]: Why was the `--replay` option removed?

If it should be removed, then you should also remove " entered previously or".

File: sub/project_settings/{{ cookiecutter.__folder_name }}/backend/Makefile [hunk 3]
```diff
@@ -102,13 +100,9 @@ update-example-content: $(BIN_FOLDER)/tox ## Export example content inside packa
 	if [ -d $(EXAMPLE_CONTENT_FOLDER)/content ]; then rm -r $(EXAMPLE_CONTENT_FOLDER)/* ;fi
 	$(BIN_FOLDER)/plone-exporter instance/etc/zope.conf $(PLONE_SITE_ID) $(EXAMPLE_CONTENT_FOLDER)
 
-.PHONY: format
-format: $(BIN_FOLDER)/tox ## Format the codebase according to our standards
+.PHONY: check
+check: $(BIN_FOLDER)/tox ## Check (and fix) codebase according to our standards
```
Comment (stevepiercy) [inline sub/project_settings/{{ cookiecutter.__folder_name }}/backend/Makefile:104]:
```suggestion
check: $(BIN_FOLDER)/tox ## Check and fix code base according to Plone standards
```

File: backend_addon/{{ cookiecutter.__folder_name }}/Makefile [hunk 4]
```diff
@@ -61,41 +59,37 @@ instance/etc/zope.ini: $(BIN_FOLDER)/pip  ## Create instance configuration
...
-install: build-dev ## Install Plone 6.0
+install: build-dev ## Install Plone
 
 .PHONY: build
-build: build-dev ## Install Plone 6.0
+build: build-dev ## Install Plone
 
 .PHONY: clean
-clean: ## Remove old virtualenv and creates a new one
+clean: ## Cleanup environment
```
Comment (stevepiercy) [inline backend_addon/{{ cookiecutter.__folder_name }}/Makefile:74]:
```suggestion
clean: ## Clean environment
```

File: backend_addon/{{ cookiecutter.__folder_name }}/Makefile [hunk 5]
```diff
@@ -61,41 +59,37 @@ instance/etc/zope.ini: $(BIN_FOLDER)/pip  ## Create instance configuration
...
 	PYTHONWARNINGS=ignore $(BIN_FOLDER)/zconsole debug instance/etc/zope.conf
 
 .PHONY: create-site
 create-site: instance/etc/zope.ini ## Create a new site from scratch
 	PYTHONWARNINGS=ignore $(BIN_FOLDER)/zconsole run instance/etc/zope.conf ./scripts/create_site.py
 
-.PHONY: format
-format: $(BIN_FOLDER)/tox ## Format the codebase according to our standards
+.PHONY: check
+check: $(BIN_FOLDER)/tox ## Check (and fix) codebase according to our standards
```
Comment (stevepiercy) [inline backend_addon/{{ cookiecutter.__folder_name }}/Makefile:91]:
```suggestion
check: $(BIN_FOLDER)/tox ## Check and fix code base according to Plone standards
```

File: sub/project_settings/{{ cookiecutter.__folder_name }}/backend/Makefile [hunk 6]
```diff
@@ -67,28 +65,28 @@ instance/etc/zope.ini: $(BIN_FOLDER)/pip  ## Create instance configuration
...
-install: build-dev ## Install Plone 6.0
+install: build-dev ## Install Plone
 
 .PHONY: build
-build: build-dev ## Install Plone 6.0
+build: build-dev ## Install Plone
 
 .PHONY: clean
-clean: ## Remove old virtualenv and creates a new one
+clean: ## Cleanup environment
```
Comment (stevepiercy) [inline sub/project_settings/{{ cookiecutter.__folder_name }}/backend/Makefile:80]:
```suggestion
clean: ## Clean environment
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
Minor grammar fixes, and a question about removing a flag.

File: project/README.md (hunk 2 above)
Comment (ericof) [reply project/README.md:54]: I plan to fix that on cookieplone

Comment (stevepiercy) [reply project/README.md:54]: Sorry, I'm not clear. What is the fix? Restore `--replay` or remove its associated text?

Comment (ericof) [reply project/README.md:54]: Restore --replay in cookieplone

File: backend_addon/tests/test_cutter_no_headless.py [hunk 7]
```diff
@@ -1,4 +1,4 @@
-"""Test cookiecutter generation with features enabled."""
+"""Test cookiecutter generation withoyt headeless feature."""
```
Comment (stevepiercy) [inline backend_addon/tests/test_cutter_no_headless.py:1]:
```suggestion
"""Test cookiecutter generation without headless feature."""
```

**Review by stevepiercy — DISMISSED** [review]

File: backend_addon/{{ cookiecutter.__folder_name }}/src/{{ cookiecutter.__package_namespace }}/{{ cookiecutter.__package_name }}/dependencies.zcml [hunk 8]
```diff
@@ -2,8 +2,8 @@
 
   <!-- List all packages you depend here -->
   {%- if cookiecutter.__feature_headless == '1' %}
-  <include package="plone.volto" />
   <include package="plone.restapi" />
```
Comment (jensens) [inline backend_addon/{{ cookiecutter.__folder_name }}/src/{{ cookiecutter.__package_namespace }}/{{ cookiecutter.__package_name }}/dependencies.zcml:5]: `plone.volto` pulls in `plone.restapi` anyway, no need to actively include it here AFAIK.

**Pull request merged**